### PR TITLE
🚀 feat: large info section

### LINKED ISSUE #339: ⚡️ 🏠 📄 ⚡️ Large Info Section component with animation
### Large Info Section

Using the [Figma Design File](https://www.figma.com/file/nLwxUWpunk8SgvjVpyE2gd/MHF-Boards?node-id=0%3A1) to put together a reusable component called `Large Info Section`. This component will have animation that brings the text into view when the component is scrolled into view using [intersection observer](https://medium.com/@cristinallamas/intersection-observer-vs-eventlistener-scroll-90aed9dc0e62).  Use [this video](https://www.youtube.com/watch?v=5DEq5cWNYt8&t=2049s) by Kevin Powell to guide you.

This info section should have:

- [ ] A boolean attribute that flips the layout to either left or right
- [ ] Animation that brings the content of the info box in from the right or left depending on the layout boolean
- [ ] A donate button that links the donate page
- [ ] A read more link that links to a program page (not yet implemented)

Screen Shot 2021-11-09 at 1 53 43 PM

### PR DESCRIPTION
**Pull request by akshitarora921**
### This PR: 
This info section have:
fixes #339
- [x]  A Boolean attribute that flips the layout to either left or right
- [ ]  Animation that brings the content of the info box in from the right or left depending on the layout Boolean
- [x]  A donate button that links the donate page
- [x]  A read more link that links to a program page (not yet implemented)

### Self-Review Checklist
- [x] PR title is clear with proper spelling and grammar
- [x] PR description contains a bulleted list of changes contained in the PR
- [x] PR links to relevant issues, with keywords to auto-close any issues fully resolved upon merge
- [x] All automated checks passed
- [x] Any variables introduced are named clearly and explicitly

### CODE REVIEW

**Comment by akshitarora921** [comment]
I have exported the Image from the figma but it contain small transparent margins.

File: components/LargeInfoSection.js [hunk 1]
```diff
@@ -0,0 +1,114 @@
+/* eslint-disable @next/next/no-img-element */
+import React, { useEffect } from 'react';
+import { motion, useAnimation } from 'framer-motion';
```
Comment (LowLifeArcade) [inline components/LargeInfoSection.js:3]: Can we do this animation without this lib using css?

File: components/LargeInfoSection.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+import useStripeSession from '../hooks/useStripeSession';
+
+function LargeInfoSection({ isLeftLayout }) {
+  const animation = useAnimation();
+  const [handleSubmit, isPending] = useStripeSession();
+  const { ref, inView } = useInView({ threshold: 0.4 });
+  const { cardsBackgroundColor, theme } = useContextTheme();
+
+  useEffect(() => {
+    const isMobile = window.innerWidth < 768;
```
Comment (LowLifeArcade) [inline components/LargeInfoSection.js:15]: Maybe we can use <style jsx> and use media queries instead of running a useEffect with an external library.

**Review by LowLifeArcade — CHANGES_REQUESTED** [review]

**Comment by LowLifeArcade** [comment]
> - [ ] Animation that brings the content of the info box in from the right or left depending on the layout Boolean

So what I should have said to be more clear is that the with the current design (which would be the default) has the image on the left and the info section on the right and that the info box should animate into view from the right. The boolean would flip the whole layout and animation would come in from the left then. 

Let's just focus on the animation at this point:
- 1. It should come in from a point on the right in a revealing fashion like [this video](https://www.youtube.com/watch?v=5DEq5cWNYt8&t=2049s) that I shared in the issue.
- 2. It should be css and come in smoothly with a fade in
- 3. The background should be dark matching the design.

File: components/LargeInfoSection.js (hunk 1 above)
Comment (akshitarora921) [reply components/LargeInfoSection.js]: Now, I am using tailwind classes

**Comment by LowLifeArcade** [comment]
> I have exported the Image from the figma but it contain small transparent margins.

Actually, the image was already in the public folder. Sorry, I should have told you that.

**Comment by akshitarora921** [comment]
> > * [ ]  Animation that brings the content of the info box in from the right or left depending on the layout Boolean
> 
> So what I should have said to be more clear is that the with the current design (which would be the default) which has the image on the left and the info section on the right, that info section and its content should animate into view from the right.
> 
> For this animation:
> 
> 1. It should come in from a point (see image below) on the right in a revealing fashion like [this video](https://www.youtube.com/watch?v=5DEq5cWNYt8&t=2049s) that I shared in the issue.
> 2. It should be css and come in smoothly with a fade in
> 3. The background should be dark matching the design.
> 
> Screen Shot 2021-12-16 at 10 28 58 AM

So, it should appear from red circle? Also should we use style jsx only or tailwind classes?

**Comment by akshitarora921** [comment]
Hi @LowLifeArcade, I have made the changes
- [x] add css using style jsx
- [x] add transition from right to left using box overlay
- [x] background color of info-box is same as of design
- [x] add a prop named `mirrored` to mirror the layout

One question: info-box color will we same for the mobile view?

File: hooks/useIntersectionObserver.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    const ob = (observer.current = new IntersectionObserver(
+      (changes) => {
+        changes.forEach((change) => {
+          if (window.innerWidth > 768) {
+            // setIsIntersecting(change.isIntersecting);
+            if (change.isIntersecting) {
+              /*
+               * 'split' turns our data attribute string into an array, and the spread operator (three dots)
+               * deconstructs it into the format that we need for manipulating the element's 'classList'
+               */
```
Comment (LowLifeArcade) [inline hooks/useIntersectionObserver.js:29]: This comment really helps 👍

**Comment by LowLifeArcade** [comment]
> Hi @LowLifeArcade, I have made the changes
> 
> * [x]  add css using style jsx
> * [x]  add transition from right to left using box overlay
> * [x]  background color of info-box is same as of design
> * [x]  add a prop named `mirrored` to mirror the layout
> 
> Questions:
> 
> 1. Info-box color will we same for the mobile view?
> 2. Should we add linear gradient to horizontal bar?

This is looking really good! As far as your questions:
1. I think that will be fine. I'd like to add some padding though. Some top and bottom and a little on the sides. Not too much but enough to give it some breathing room.
2. Hmm, I'd prefer not to use a gradient. BUT I'd like it to animate! Let's make it animate into frame from the right to the left as the content fades in. Basically do what you did in the info cards in the card carousel.

**Comment by akshitarora921** [comment]
> > Hi @LowLifeArcade, I have made the changes
> > 
> > * [x]   add css using style jsx
> > * [x]   add transition from right to left using box overlay
> > * [x]   background color of info-box is same as of design
> > * [x]   add a prop named `mirrored` to mirror the layout
> > 
> > Questions:
> > 
> > 1. Info-box color will we same for the mobile view?
> > 2. Should we add linear gradient to horizontal bar?
> 
> This is looking really good! As far as your questions:
> 
> 1. I think that will be fine. I'd like to add some padding though. Some top and bottom and a little on the sides. Not too much but enough to give it some breathing room.
> 2. Hmm, I'd prefer not to use a gradient. BUT I'd like it to animate! Let's make it animate into frame from the right to the left as the content fades in. Basically do what you did in the info cards in the card carousel.

@LowLifeArcade, I have completed the changes, Please take a look.

Changes-
- [ ] Add more padding to info-box in mobile view
- [ ] Add width transition on horizontal bar

**Comment by LowLifeArcade** [comment]
@akshitarora921 this is great! Well done addressing all of the comments and making all of the adjustments. I'll talk to you soon, buddy. Merging.

**Pull request merged**